### ISSUE TITLE
Android inference issue  Label number 1 mismatch the shape on axis 1

### ISSUE THREAD

**Issue by NazaRik555** [ultralytics/ultralytics#15691]
### Search before asking

- [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### Ultralytics YOLO Component

Export

### Bug

I'm trying to use default pre-trained yolov8n model to make detection on Android. I've used export to tflite with int8=true.
Loaded my model into android studio and then try to run
```
DetectorBestInt8 model = DetectorBestInt8.newInstance(getApplicationContext());

// Creates inputs for reference.
TensorImage imageTensor = TensorImage.fromBitmap(image);

// Runs model inference and gets result.
DetectorBestInt8.Outputs outputs = model.process(imageTensor);

List<Category> output = outputs.getOutputAsCategoryList();
```

but at the last line my app crashes with error.

Here is my LogCat 
```
FATAL EXCEPTION: main
            Process: com.example.carclf, PID: 15280
            java.lang.IllegalArgumentException: Label number 1 mismatch the shape on axis 1
             at org.tensorflow.lite.support.common.SupportPreconditions.checkArgument(SupportPreconditions.java:104)
             at org.tensorflow.lite.support.label.TensorLabel.<init>(TensorLabel.java:87)
             at org.tensorflow.lite.support.label.TensorLabel.<init>(TensorLabel.java:105)
             at com.example.carclf.ml.DetectorBestInt8$Outputs.getOutputAsCategoryList(DetectorBestInt8.java:105)
             at com.example.carclf.TestModelActivity.detectImage(TestModelActivity.java:189)
             at com.example.carclf.TestModelActivity.lambda$new$1$com-example-carclf-TestModelActivity(TestModelActivity.java:252)
             at com.example.carclf.TestModelActivity$$ExternalSyntheticLambda0.onActivityResult(D8$$SyntheticClass:0)
             at androidx.activity.result.ActivityResultRegistry$1.onStateChanged(ActivityResultRegistry.java:147)
             at androidx.lifecycle.LifecycleRegistry$ObserverWithState.dispatchEvent(LifecycleRegistry.kt:314)
             at androidx.lifecycle.LifecycleRegistry.forwardPass(LifecycleRegistry.kt:251)
             at androidx.lifecycle.LifecycleRegistry.sync(LifecycleRegistry.kt:287)
             at androidx.lifecycle.LifecycleRegistry.moveToState(LifecycleRegistry.kt:136)
             at androidx.lifecycle.LifecycleRegistry.handleLifecycleEvent(LifecycleRegistry.kt:119)
             at androidx.lifecycle.ReportFragment$Companion.dispatch$lifecycle_runtime_release(ReportFragment.kt:192)
             at androidx.lifecycle.ReportFragment$LifecycleCallbacks.onActivityPostStarted(ReportFragment.kt:121)
             at android.app.Activity.dispatchActivityPostStarted(Activity.java:1362)
             at android.app.Activity.performStart(Activity.java:8061)
             at android.app.ActivityThread.handleStartActivity(ActivityThread.java:3475)
             at android.app.servertransaction.TransactionExecutor.performLifecycleSequence(TransactionExecutor.java:221)
             at android.app.servertransaction.TransactionExecutor.cycleToPath(TransactionExecutor.java:201)
             at android.app.servertransaction.TransactionExecutor.executeLifecycleState(TransactionExecutor.java:173)
             at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:97)
             at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2066)
             at android.os.Handler.dispatchMessage(Handler.java:106)
             at android.os.Looper.loop(Looper.java:223)
             at android.app.ActivityThread.main(ActivityThread.java:7656)
             at java.lang.reflect.Method.invoke(Native Method)
             at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:592)
             at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:947)
```

### Environment

-

### Minimal Reproducible Example

-

### Additional

-

### Are you willing to submit a PR?

- [x] Yes I'd like to help by submitting a PR!

**Comment by glenn-jocher** [comment]
Thank you for reporting this issue. Please ensure you are using the latest version of the Ultralytics package. If the problem persists, it might be related to the label file or model input/output dimensions. Double-check that the label file matches the model's output classes and that the input image dimensions are correct. If the issue continues, please provide more details about your setup.

**Comment by glenn-jocher** [comment]
@NazaRik555 it looks like the issue might be due to the absence of a label file. Please ensure you include a label file that matches the model's output classes. This should resolve the mismatch error you're encountering.

**Comment by NazaRik555** [comment]
> output

Would it load automatically or I should do it manually?

**Comment by glenn-jocher** [comment]
@NazaRik555 the label file should be loaded automatically if it's included in the model's assets. If not, you'll need to load it manually.

**Comment by FishWaterUnder** [comment]
I have the same problem here . I am using yolov8 pose in android studio with tflite and it also shows the same error.

Here is my code 
```
bitmap = textureView.bitmap!!
var tensorImage = TensorImage(DataType.FLOAT32)
tensorImage.load(bitmap)

tensorImage = imageProcessor.process(tensorImage)

val inputFeature0 = TensorBuffer.createFixedSize(intArrayOf(1, 640, 640, 3), DataType.FLOAT32)
inputFeature0.loadBuffer(tensorImage.buffer)

val outputs = model.process(inputFeature0)
val outputFeature0 = outputs.outputAsCategoryList

Log.i("Hello",outputFeature0.toString())
```

I used the pretrained Yolov8 pose model but I don't know where the labels are. I tried it already on python without having to use any labels. what should I do?

**Comment by glenn-jocher** [comment]
@FishWaterUnder it seems the issue is related to the absence of a label file. YOLOv8 models typically require a label file to map the output classes correctly. Please ensure you include a label file that matches the model's output classes in your Android project. If the issue persists, verify that you are using the latest version of the Ultralytics package.

**Comment by FishWaterUnder** [comment]
@glenn-jocher where can I find the label file of the pretrained model?

I followed the example in the website when exporting the tflite file but it doesn't give me a label file.

**Comment by glenn-jocher** [comment]
You can manually create a label file by listing the class names in the same order as they appear in the model's training dataset. Save this file as a `.txt` file and include it in your Android project.

**Comment by FishWaterUnder** [comment]
@glenn-jocher is there a specific name for the `.txt` file? and are the labels found in the metadata? in this case is it this?:

```
names:
  0: person
```

**Comment by glenn-jocher** [comment]
The `.txt` file should contain the class names, one per line, in the same order as they appear in the model's training dataset. You can name the file `labels.txt` and include it in your Android project. Yes, the labels are typically found in the metadata under the `names` section.
